.circleci/config.yml

```diff
@@ -851,33 +801,46 @@ workflows:
                 - /^rc.*/
                 - /^hotfix.*/
 
-  launch-all-app-builds-workflow:
-    # using do-builds param to select which of the two workflows using the build-on-tag-filters
-    # should run; should be this one that just launches the builds or the one actually doing the builds
-    unless: << pipeline.parameters.do-builds >>
-    jobs:
-      - conditionally-launch-build-and-store-all-job: &build-and-store-filters
-          filters:
-            branches:
-              only:
-                - /^rc.*/
-                - /^hotfix.*/
-
   build-app-workflow:
     when: << pipeline.parameters.do-builds >>
     jobs:
       - ui-unit-test:
-          <<: *build-and-store-filters
+          <<: *filter-rc-hotfix-branch
           study_key: << pipeline.parameters.study_key >>
       - api-unit-test:
-          <<: *build-and-store-filters
+          <<: *filter-rc-hotfix-branch
       - app-build:
-          <<: *build-and-store-filters
+          <<: *filter-rc-hotfix-branch
           study_key: << pipeline.parameters.study_key >>
           requires:
             - ui-unit-test
             - api-unit-test
 
+  build-apps-rc-hotfix:
```

Comment:
So the new workflow will address issues found in `launch-all-app-builds-workflow` workflow.

- Run sdk and tooltit unit tests one time. Not repeated for every study in `build-app-workflow` workflow.
- For every study, build will require ui unit tests in that study to pass first.
- For every study, build will only happen if it has been changes found in `rc_*` or `hotfix_*` branches.
- Job fails then workflow fails too.
- matrix is used to reduce code clutter.